tools/zygote/run_gmm.jl

```diff
@@ -0,0 +1,91 @@
+module GMM
+
+import Zygote
+import GradBench
+
+function unpack(input)
+    gmm_input = GradBench.GMM.input_from_json(input)
+
+    d = size(gmm_input.x, 1)
+    k = size(gmm_input.means, 2)
+    Qs = cat([GradBench.GMM.get_Q(d, gmm_input.icfs[:, ik]) for ik in 1:k]...;
+             dims=[3])
+
+    return (gmm_input.alphas, gmm_input.means, Qs, gmm_input.x, gmm_input.wishart)
+end
+
+Zygote.@adjoint function GradBench.GMM.diagsums(Qs)
+    GradBench.GMM.diagsums(Qs),
+    function (Δ)
+        Δ′ = zero(Qs)
+        for (i, δ) in enumerate(Δ)
+            for j in 1:size(Qs, 1)
+                Δ′[j, j, i] = δ
+            end
+        end
+        (Δ′,)
+    end
+end
+
+Zygote.@adjoint function GradBench.GMM.expdiags(Qs)
+    GradBench.GMM.expdiags(Qs),
+    function (Δ)
+        Δ′ = zero(Qs)
+        Δ′ .= Δ
+        for i in 1:size(Qs, 3)
+            for j in 1:size(Qs, 1)
+                Δ′[j, j, i] *= exp(Qs[j, j, i])
+            end
+        end
+        (Δ′,)
+    end
+end
+
+Zygote.@adjoint function Base.map(f, args...)
```

Comment:
I am surprised that we need a custom adjoint for map here

Reply:
As am I. It was part of the original ADBench code, but it seems to work fine without it, so I will remove it.